MudAutocomplete: `ItemDisabledTemplate` and `ItemSelectedTemplate` will now display even if `ItemTemplate` is not defined.

digitaldirk opened the pull request:
## Description
Updated the logic for choosing which template (or just the string) for an item. Previously if the ItemTemplate was null, then ItemDisabledTemplate/ItemSelectedTemplate would not display.
Fixes #8912 

## How Has This Been Tested?
Visually tested via docs and own project.

## Type of Changes
Would this be a visually breaking bug? If someone defined just a ItemDisabledTemplate and no others, before this fix their MudAutocomplete would be "normal" and after this fix then the disabled items could look different. This approach also prioritizes the templates in this order: ItemDisabledTemplate -> ItemSelectedTemplate -> ItemTemplate -> item string. **An example to think about:** if all these templates were defined, and programmatically we selected an item that is marked as Disabled in a MudAutocomplete then how should it show? I believe disabled is the best choice here but would love to hear opinions about this.
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (fix or improvement to the website or code docs)

**Docs demo:**
![DocsItemDisabledTemplateDemo](https://github.com/MudBlazor/MudBlazor/assets/22691956/73e391c2-80df-431b-b8b8-8ef03b64bda8)

**Current behavior:**
You can see how the ItemDisabledTemplate does not display when the ItemTemplate is null here: https://try.mudblazor.com/snippet/GkmIuTaLryaVxwKA

## Checklist
- [X] The PR is submitted to the correct branch (`dev`).
- [X] My code follows the code style of this project.
- [ ] I've added relevant tests.

digitaldirk committed:
Fix for ItemDisabledTemplate/ItemSelectedTemplate not showing when ItemTemplate was null

Previously when ItemTemplate was null, ItemDisabledTemplate/ItemSelectedTemplate would not show. Now ItemDisabledTemplate takes priority, then ItemSelectedTemplate, then ItemTemplate, finally just the string if all are null.

digitaldirk committed:
Updated presentation section with ItemDisabledTemplate example

Shows how to use the ItemDisabledTemplate to customize a disabled item.

digitaldirk committed:
Updated Presentation description to included ItemDisabledTemplate

Reworded text a little, and added ItemDisabledTemplate.

ScarletKuro commented:
Hi.
Thanks for your PR.
Can you please add a bUnit test so we will be sure that https://github.com/MudBlazor/MudBlazor/issues/8912 won't happen again

ScarletKuro commented:
@danielchalmers probs will conflict with https://github.com/MudBlazor/MudBlazor/pull/8914 as same line was moddified

digitaldirk committed:
Update AutocompleteTests.cs to include ItemTemplateDisplay test

Tests that the ItemDisabledTemplate and ItemSelectedTemplate both can display when ItemTemplate isn't provided (null)

digitaldirk committed:
Add Autocomplete Item Template Display Test component

For use with "AutocompleteItemTemplateDisplayTest" Test in AutocompleteTests.cs

digitaldirk commented:
Test is now added @henon @danielchalmers @ScarletKuro 

Test passes on my branch and fails on dev. The test component has both the `ItemDisabledTemplate` and `ItemSelectedTemplate` defined and it does not have a `ItemTemplate` defined

**On my branch:**
![image](https://github.com/MudBlazor/MudBlazor/assets/22691956/79c89d80-1067-41e7-9e72-75b331f1bbb0)

**On dev:**
![image](https://github.com/MudBlazor/MudBlazor/assets/22691956/2c84d688-d95b-4ed6-9122-f0cdef5392d3)

digitaldirk committed:
Merge branch 'dev' into MudAutocomplete-Disabled/Selected-Templates-Fix

henon commented:
@digitaldirk what would you write in the migration guide?

digitaldirk commented:
> @digitaldirk what would you write in the migration guide? Just a one-liner please

`ItemDisabledTemplate` and `ItemSelectedTemplate` will now display when a `ItemTemplate` is not defined.

@henon

henon commented:
Thanks @digitaldirk 

Added to v7.0.0 Migration Guide #8447
